provision.yml

```
@@ -0,0 +1,88 @@
+---
+- name: configure environment
+  hosts: localhost
+  connection: local
+  vars_files:
+    - group_vars/couchbase/vars
+    - group_vars/couchbase/secret_vars
+
+  vars: 
+    count: '{{ servers - 1 }}' # udf 
+    
+  tasks: 
+  - name: check minumum node size
+    fail: 
+      msg: "CouchBase Server requires a minumum of 8GB RAM"
+    when: (type == 'g6-nanode-1') or 
+          (type == 'g6-standard-1') or 
+          (type == 'g6-standard-2') or 
+          (type == 'g6-dedicated-2') or 
+          (type == 'g7-premium-2') | bool
+
+  - name: create couchbase cluster
+    linode.cloud.instance: 
+      label: '{{ instance_prefix }}-{{ item | int +1 }}-{{ uuid }}'
+      api_token: '{{ api_token }}'
+      type: '{{ type }}'
+      region: '{{ region }}'
+      image: '{{ image }}'
+      root_pass: '{{ root_pass }}'
+      authorized_keys: '{{ ssh_keys }}'
+      private_ip: true
+      stackscript_id: 1366193
+      ua_prefix: 'marketplace-couchbase-occ'
+      tags: '{{ cluster_uuid }}'
+      state: present
+    with_sequence: count='{{ count }}'
+
+  - name: collect cluster details
+    linode.cloud.instance_info:
+      api_token: '{{ api_token }}'
+      label: '{{ instance_prefix }}-{{ item | int }}-{{ uuid }}'
+    register: cluster_info
+    with_sequence: count='{{ count | int + 1 }}'
+
+  - name: update group_vars with server variables
+    blockinfile:
+      path: ./group_vars/couchbase/vars
+      marker: "# {mark} INSTANCE VARS"
+      block: |
+        #jinja2: trim_blocks:False
+        couchbase_data:
+          server:
+            {%- for count in range(count | int + 1) %}
+            - couchbase-server-{{ count }}:
+              instance:
+                hostname: {{ cluster_info.results[count].instance.label | replace('_', '-') }}
+                ip_pub1: {{ cluster_info.results[count].instance.ipv4[0] }}
+                ip_priv1: {{ cluster_info.results[count].instance.ipv4[1] }}
+            {%- endfor %}        
```

```suggestion
            - couchbase-server-{{ count | int + 1}}:
              instance:
                hostname: {{ cluster_info.results[count].instance.label | replace('_', '-') }}
                ip_pub1: {{ cluster_info.results[count].instance.ipv4[0] }}
                ip_priv1: {{ cluster_info.results[count].instance.ipv4[1] }}
            {%- endfor %} 
```

What you have produces:
```
couchbase_data:
  server:
    - couchbase-server-0:
      instance:
        hostname: cassandra-1-eab34182
        ip_pub1: xxxx
        ip_priv1: xxxx
    - couchbase-server-1:
      instance:
        hostname: cassandra-2-eab34182
        ip_pub1: xxxx
        ip_priv1: xxxx
    - couchbase-server-2:
      instance:
        hostname: cassandra-3-eab34182
        ip_pub1: xxxx
        ip_priv1: xxxx
 (..snip)  
```

---

I decided to add the int so hostnames and external Linode labels match. The Ansible inv name of the instance won't be visible to the end-user.